src/BaseDAO/ShareTest/Token.hs

```diff
@@ -0,0 +1,90 @@
+-- SPDX-FileCopyrightText: 2020 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+module BaseDAO.ShareTest.Token
+  ( adminBurnScenario
+  , adminMintScenario
+  , transferContractTokensScenario
+  ) where
+
+import Universum
+
+import Lorentz hiding ((>>))
+import Morley.Nettest
+import Util.Named
+
+import qualified Lorentz.Contracts.BaseDAO.Types as DAO
+import qualified Lorentz.Contracts.Spec.FA2Interface as FA2
+import BaseDAO.ShareTest.Common
+
+{-# ANN module ("HLint: ignore Reduce duplication" :: Text) #-}
+
+
+-- TODO: Add test to ensure only admin can burn/mint
```

Comment:
As well as `transfer_contract_tokens`, I think this should be done in this issue already.

Reply:
Yeah, I think so, I have added the tests now.